apps/keyboard/js/keyboard/user_press_manager.js

```diff
@@ -234,11 +256,73 @@ UserPressManager.prototype._handleFinishPress = function(el, coords, id) {
   press.updateCoords(coords,
     press.moved || this._distanceReachesLimit(id, coords));
 
-  if (typeof this.onpressend === 'function') {
-    this.onpressend(press, id);
+  if(this._lowEndDevice && this._exceedSpeedLimit(press) &&
```

Comment:
Moved the code for low end devices to a separate function. Also initialize the `_isLowEndDevice` during startup like you said.